public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -428,30 +434,31 @@ function rulerRuleToCombinedRule(
           group,
           instanceTotals: {},
           filteredInstanceTotals: {},
+          uid: rulerRuleType.grafana.rule(rule) ? rule.grafana_alert.uid : undefined,
         };
 }
 
 // find existing rule in group that matches the given prom rule
 function getExistingRuleInGroup(
   rule: Rule,
-  existingCombinedRulesMap: Map<string, CombinedRule[]>,
+  existingCombinedRulesMap: Map<string, CombinedRule[]>, // map by uid
   rulesSource: RulesSource
 ): CombinedRule | undefined {
   // Using Map of name-based rules is important performance optimization for the code below
   // Otherwise we would perform find method multiple times on (possibly) thousands of rules
 
-  const nameMatchingRules = existingCombinedRulesMap.get(rule.name);
-  if (!nameMatchingRules) {
+  const fieldToSearch = getIdToCompare(rule);
+  const uidOrNameMatchingRules = existingCombinedRulesMap.get(fieldToSearch ?? '');
```

Comment:
nit: do we need the nullish here? IIRC `getIdToCompare` doesn't return undefined?

Reply:
true!